### PR TITLE
fix vectorsqrt

### PR DESCRIPTION
**Pull request by glacambre**
Fix vectorSqrt example

### CODE REVIEW

File: examples/0_Simple/vectorSqrt/device.gpr [hunk 1]
```diff
@@ -11,9 +11,16 @@ project Device is
...
+   for Library_Name use "kernel";
+   for Library_Dir use "lib";
+
+   for Archive_Builder use ("cuda-gcc", "-v");
+   for Archive_Indexer use ("");
+   for Archive_Suffix use ".fatbin.o";
+
    package Compiler is
-      for Switches ("ada") use ("-O2", "-gnatn", "-nostdlib");
+      for Switches ("ada") use ("-gnatX", "-O2", "-gnatp", "-gnatn", "-v");
```
Comment (leogermond) [inline examples/0_Simple/vectorSqrt/device.gpr:22]: it seems that the flags changed?
rm -nostdlib
add -gnatp, -gnatX, and -v

Are they all expected?

File: examples/0_Simple/vectorSqrt/device.gpr [hunk 2]
```diff
@@ -11,9 +11,16 @@ project Device is
 
    for Create_Missing_Dirs use "True";
 
+   for Library_Name use "kernel";
+   for Library_Dir use "lib";
+
+   for Archive_Builder use ("cuda-gcc", "-v");
```
Comment (leogermond) [inline examples/0_Simple/vectorSqrt/device.gpr:17]: Could `-v` be part of `package Builder`?

**Review by leogermond — DISMISSED** [review]

File: examples/0_Simple/vectorSqrt/device.gpr (hunk 1 above)
Comment (glacambre) [reply examples/0_Simple/vectorSqrt/device.gpr:22]: Right, `-nostdlib` is actually harmful (we want to build against the stdlib, which will contain necessary symbols) and was a leftover from previous experiments.

`-O2`, `-gnatp` and `-gnatn` are perf flags.

`-v` helps with debugging issues, should they arise.

File: examples/0_Simple/vectorSqrt/device.gpr (hunk 2 above)
Comment (glacambre) [reply examples/0_Simple/vectorSqrt/device.gpr:17]: I don't know, this was copy/pasted from another gprfile Quentin wrote. I'll look into whhat using `-v` for `Builder` entails, thanks for the suggestion :)

Comment (leogermond) [reply examples/0_Simple/vectorSqrt/device.gpr:17]: From what I can gather it applies as default flag to Compiler, Binder, and Linker, which seems good for the `-v` flag

Comment (glacambre) [reply examples/0_Simple/vectorSqrt/device.gpr:17]: What would be the right incantation? I tried
```ada
   package Builder is
      for Default_Switches ("Ada") use ("-v");
      -- for Default_Switches ("main") use ("-v");
      -- for Default_Switches ("gnatcuda_wrapper.adb") use ("-v");
      -- for Switches ("Ada") use ("-v");
      -- for Switches ("main") use ("-v");
      -- for Switches ("gnatcuda_wrapper.adb") use ("-v");
   end Builder;
```
And nothing worked.

Comment (QuentinOchem) [reply examples/0_Simple/vectorSqrt/device.gpr:17]: You don't need Archive_Builder anymore - I moved these in the gprbuild database with Vasiliy at the begining of the week. You can remove Archive_Indexer, Archive_Suffix as well as the Object_File_Suffix of the compiler package as well - as long as you have a recent enough toolchain installed. 

I was planning to update all examples with that as soon as I have builds, which should have been today ;-)

**Review by QuentinOchem — APPROVED** [review]

**Pull request merged**